airflow/providers/amazon/aws/sensors/base_aws.py

```diff
@@ -0,0 +1,145 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
All of he above suggestions apply down here as well.

Reply:
Could we make BaseSensor inherit BaseOperator and save a lot of duplicate code?   A Sensor is just a specialized Operator, really.

Reply:
This might works 

```python
class AwsBaseSensor(AwsBaseOperator, BaseSensorOperator)
    ...
```

However we would step into [MRO](https://www.python.org/download/releases/2.3/mro/) field, if we do something wrong and forget about some important thing, it might easily cause an issue in multiple sensor eventually.

Reply:
Just for comparison, seems like it should work `BaseSensorOperator` use before `BaseOperator` however if it not strict concern to DRY and use this kind of multi inheritance I would rather keep it with bunch of repeatable code. At least until we do not move existed AWS Sensors to AwsBaseSensor

MRO: class AwsBaseSensor(AwsBaseOperator, BaseSensorOperator)
```
1:	<class 'airflow.providers.amazon.aws.sensors.base_aws.AwsBaseSensor'>
2:	<class 'airflow.providers.amazon.aws.operators.base_aws.AwsBaseOperator'>
3:	<class 'airflow.sensors.base.BaseSensorOperator'>
4:	<class 'airflow.models.baseoperator.BaseOperator'>
5:	<class 'airflow.models.abstractoperator.AbstractOperator'>
6:	<class 'airflow.template.templater.Templater'>
7:	<class 'airflow.models.skipmixin.SkipMixin'>
8:	<class 'airflow.utils.log.logging_mixin.LoggingMixin'>
9:	<class 'airflow.models.taskmixin.DAGNode'>
10:	<class 'airflow.models.taskmixin.DependencyMixin'>
11:	<class 'typing.Generic'>
12:	<class 'object'>
```

MRO: class AwsBaseSensor(BaseSensorOperator, Generic[_AwsHook]):
```
1:	<class 'airflow.providers.amazon.aws.sensors.base_aws.AwsBaseSensor'>
2:	<class 'airflow.sensors.base.BaseSensorOperator'>
3:	<class 'airflow.models.baseoperator.BaseOperator'>
4:	<class 'airflow.models.abstractoperator.AbstractOperator'>
5:	<class 'airflow.template.templater.Templater'>
6:	<class 'airflow.models.skipmixin.SkipMixin'>
7:	<class 'airflow.utils.log.logging_mixin.LoggingMixin'>
8:	<class 'airflow.models.taskmixin.DAGNode'>
9:	<class 'airflow.models.taskmixin.DependencyMixin'>
10:	<class 'typing.Generic'>
11:	<class 'object'>
```

Reply:
Yeah, I think you are right.  Leave it as you have it, I guess.  Thanks for considering it though.